### PR TITLE
[Feature][Connector-V2] Add support for XML file type to various file connectors

### PR DESCRIPTION
**Pull request by Ivan-gfan**
Add support for XML file type to various file connectors such as SFTP, FTP, LocalFile, HdfsFile, and more.  #6326

### CODE REVIEW

**Comment by Ivan-gfan** [comment]
#6327

**Comment by Hisoka-X** [comment]
cc @TyrantLucifer

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-sftp-e2e/src/test/resources/xml/e2e.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-sftp-e2e/src/test/resources/xml/e2e.xml:1]: Why is this file needed? You can first create xml through sink and then read xml through source

Comment (Ivan-gfan) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-sftp-e2e/src/test/resources/xml/e2e.xml:1]: At first, I was using this file for source, but later I found out that I could generate it using fake source, and then read it again. I did not delete the file.

**Review by EricJoy2048 — APPROVED** [review]

**Review by liugddx — APPROVED** [review]
lgtm
